src/wp-includes/class-wp-scripts.php

```diff
@@ -333,7 +333,12 @@ public function do_item( $handle, $group = false ) {
 			 */
 			$srce = apply_filters( 'script_loader_src', $src, $handle );
 
-			if ( $this->in_default_dir( $srce ) && ( $before_handle || $after_handle || $translations_stop_concat ) ) {
+			// Get the most eligible loading strategy for said script handle.
+			// Used as a conditional to prevent script concatenation.
+			$strategy                    = $this->get_eligible_loading_strategy( $handle );
+			$is_deferred_or_async_handle = '' !== $strategy;
```

Comment:
maybe check explicitly for 'async' or defer' here since other strategies could be added in the future.